### PR TITLE
Make some Image methods static

### PR DESCRIPTION
**Pull request by KoBeWi**
Follow-up to https://github.com/godotengine/godot/pull/60739#issuecomment-1192357150

### CODE REVIEW

File: core/io/image.cpp [hunk 1]
```diff
@@ -2051,6 +2051,24 @@ Vector<uint8_t> Image::get_data() const {
...
+	image.instantiate();
+	image->create(p_width, p_height, p_use_mipmaps, p_format, p_data);
+	return image;
+}
+
+void Image::set_data(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data) {
+	create(p_width, p_height, p_use_mipmaps, p_format, p_data);
+}
+
 void Image::create(int p_width, int p_height, bool p_use_mipmaps, Format p_format) {
```
Comment (akien-mga) [inline core/io/image.cpp:2072]: I think this warrants further refactoring, it's weird to have static `create_empty` and `create_from_data` and non-static `create` with two variants.

Maybe:
- Move `create(int p_width, int p_height, bool p_use_mipmaps, Format p_format)` logic to `create_empty` (and rename to `create` to match bindings)
- Move `create(int p_width, int p_height, bool p_use_mipmaps, Format p_format, const Vector<uint8_t> &p_data)` logic to `set_data` and use it in `create_from_data`.

**Review by reduz — DISMISSED** [review]

**Comment by reduz** [comment]
looks ok to me

**Comment by akien-mga** [comment]
> I misunderstood what you were doing. I am not entirely sure its good to make these static, since they are operating on the image itself.

It's only the `create` methods which are static, I think that makes sense? It creates and give you a new Image.

**Comment by KoBeWi** [comment]
I think the internal `create()` method could be renamed to `create_data()`, because that's what it actually does.

**Comment by KoBeWi** [comment]
I renamed the internal `create()` to `initialize_data()` (I didn't use `create_data()` to avoid confusion with `create_from_data()`) and also replaced remaining old calls with static methods where applicable.

File: core/io/image.cpp [hunk 2]
```diff
@@ -3462,7 +3479,7 @@ Ref<Image> Image::rgbe_to_srgb() {
 
 	Ref<Image> new_image;
 	new_image.instantiate();
-	new_image->create(width, height, false, Image::FORMAT_RGB8);
+	new_image->initialize_data(width, height, false, Image::FORMAT_RGB8);
```
Comment (akien-mga) [inline core/io/image.cpp:3482]: Could use `create_empty()` to streamline the code.

**Review by akien-mga — APPROVED** [review]
We discussed this in a PR meeting.

The change seems good, but we're concerned that it will lead to more problematic situations where users are unaware that their code is no longer working when using the previous format (which would still work silently).

We should find a proper solution to this UX issue in GDScript before merging more such changes that will impact users of the beta version.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**